### PR TITLE
Enable the topic operator's ZK-based metadata store unit test

### PR DESCRIPTION
**Pull request by fvaleri**
This is still useful because you can enable the ZK based store by setting STRIMZI_USE_ZOOKEEPER_TOPIC_STORE.

### CODE REVIEW

**Comment by fvaleri** [comment]
It works on my machine, but it looks like it is failing on CI/CD. Will look at it later.

**Comment by fvaleri** [comment]
This is now ready for review. If it's fine with you, I would leave the `getFreePort` small refactoring, which is unrelated to the fix.

Output from CI/CD:

```
[INFO] Running io.strimzi.operator.topic.ZkTopicStoreTest
2023-04-05 13:41:18 [ZOOKEEPER] WARN  ServerCnxnFactory:309 - maxCnxns is not configured, using default value 0.
2023-04-05 13:41:18 INFO  ZkEventThread:65 - Starting ZkClient event thread.
2023-04-05 13:41:18 INFO  ZkClient:936 - Waiting for keeper state SyncConnected
2023-04-05 13:41:18 INFO  ZkClient:713 - zookeeper state changed (SyncConnected)
2023-04-05 13:41:19 INFO  ZkEventThread:83 - Terminate ZkClient event thread.
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.784 s - in io.strimzi.operator.topic.ZkTopicStoreTest
```

File: topic-operator/src/test/java/io/strimzi/operator/topic/ZkTopicStoreTest.java [hunk 1]
```diff
@@ -14,18 +16,17 @@
...
     private static Vertx vertx;
-
     private EmbeddedZooKeeper zkServer;
-    private Zk zk;
+    private Zk zkClient;
 
     @Override
     protected boolean canRunTest() {
-        return false; // test was disabled
+        return true;
```
Comment (scholzj) [inline topic-operator/src/test/java/io/strimzi/operator/topic/ZkTopicStoreTest.java:29]: Is this `canRunTest` construct still needed?

**Review by scholzj — COMMENTED** [review]
If we mix the `getFreePort` here => can you please use it also in `ConnectCluster`? Or is there some reason why it is not possible?

File: topic-operator/src/test/java/io/strimzi/operator/topic/ZkTopicStoreTest.java (hunk 1 above)
Comment (fvaleri) [reply topic-operator/src/test/java/io/strimzi/operator/topic/ZkTopicStoreTest.java:29]: I don't think so, let me get rid of it.

**Comment by fvaleri** [comment]
> If we mix the `getFreePort` here => can you please use it also in `ConnectCluster`? Or is there some reason why it is not possible?

I simply forgot about it. Thanks.

**Review by scholzj — APPROVED** [review]
LGTM. It seems to be now passing locally for me as well. Thanks.

**Review by tombentley — APPROVED** [review]
Thanks @fvaleri

**Pull request merged**